### ISSUE TITLE
Connection loss after each CGM value refresh

### ISSUE THREAD

**Issue by flavi0gritti** [janvv/GarminService#3]
As suggested on Zulip I'm opening this issue because after building the Loop app with Garmin service and importing my garmin devices (Forerunner 965 and an Edge 1040) I can't get real time readings on any of them.

Basically after connecting the device inside the Garmin service and starting an activity on the device the complication shows only one CGM value before losing sync and connection to my smartphone (showing either past value or after roughly 10 minutes 3 loading dots instead of a number).

Tested both on my 965 and 1040 as shown in the video attached.

https://github.com/user-attachments/assets/ca098470-061c-46ab-9b7d-3b5f72e47fb0

https://github.com/user-attachments/assets/5820f7be-f368-4e76-b40e-749e851dd843

I'm trying to gather some log files from xcode while the phone is still plugged in but I am having a difficult time figuring out how to do that.

**Comment by micro17** [comment]
Hey Flavio welcome 
Which cgm are you using?

And could you provide thexcode  log file while connecting / when a cgm value comes in?
There you can follow live if it tries to send a value and if it succeeds.

**Comment by janvv** [comment]
To deploy the app on your phone you need to click the "Play" Button in Xcode (top left). This will compile the app and run it on your phone while Xcode is still connected to it. Xcode will then log all the debug messages. When you disconnect the app from Xcode (plugging the cable or killing the app), Xcode is no longer connected to it.

So everytime you want to see the debug messages, you'll need to re-deploy the app using the play button. This does not delete your Loop settings so it is typically not a problem.

Afterwards you should be able to find the entry (try the search if you cant find it)
![xcode1](https://github.com/user-attachments/assets/9397a151-ae37-479f-8a38-26648510e6b4)
![xcode2](https://github.com/user-attachments/assets/16237265-2657-49c3-8d57-4d920373254b)
![xcode3](https://github.com/user-attachments/assets/3f042320-daf2-414f-ba2f-6350659c9fd6)

**Comment by flavi0gritti** [comment]
> Hey Flavio welcome Which cgm are you using?
> 
> And could you provide thexcode log file while connecting / when a cgm value comes in? There you can follow live if it tries to send a value and if it succeeds.

Dexcom G6, soon to be G7

**Comment by flavi0gritti** [comment]
```
ConnectIQ initialized
Type: Debug | Timestamp: 2024-10-08 08:36:11.054396+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
File: com.apple.watchconnectivity
Type: Debug | Timestamp: 2024-10-08 08:36:11.072015+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: DeviceManager | TID: 0x1c8ac8
File: com.JJ2G92ZDC7.loopkit.Loop
Type: Debug | Timestamp: 2024-10-08 08:36:11.072068+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: DeviceManager | TID: 0x1c8ac8
File: devices
Type: Debug | Timestamp: 2024-10-08 08:36:11.072091+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: DeviceManager | TID: 0x1c8ac8
Garmin DeviceManager: Restored saved devices:
Type: Debug | Timestamp: 2024-10-08 08:36:11.074342+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: DeviceManager | TID: 0x1c8ac8
<IQDevice: 0x2350aa0, uuid = D5F828F2-A31D-6B64-CF8C-FBAAC7EB0C68, modelName = "Edge 1040", friendlyName = "Edge 1040">
Type: Debug | Timestamp: 2024-10-08 08:36:11.074421+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: DeviceManager | TID: 0x1c8ac8
<IQDevice: 0x2353ea0, uuid = 25919FAB-4F89-F299-47A7-F419605354AF, modelName = "Forerunner 965", friendlyName = "Forerunner 965">
Type: Debug | Timestamp: 2024-10-08 08:36:11.074454+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: DeviceManager | TID: 0x1c8ac8
Devices changed: [<IQDevice: 0x2350aa0, uuid = D5F828F2-A31D-6B64-CF8C-FBAAC7EB0C68, modelName = "Edge 1040", friendlyName = "Edge 1040">, <IQDevice: 0x2353ea0, uuid = 25919FAB-4F89-F299-47A7-F419605354AF, modelName = "Forerunner 965", friendlyName = "Forerunner 965">]
Type: Info | Timestamp: 2024-10-08 08:36:11.074506+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
GarminServiceTableViewController: Button added
Type: Debug | Timestamp: 2024-10-08 08:36:11.084363+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
GarminServiceTableViewController: Button added
Type: Debug | Timestamp: 2024-10-08 08:36:11.094400+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
Deregistering for device events for <IQDevice: 0x23ee7a0, uuid = D5F828F2-A31D-6B64-CF8C-FBAAC7EB0C68, modelName = "", friendlyName = "">
Type: Info | Timestamp: 2024-10-08 08:36:12.559491+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c8ac8
Setting active garmin device to <IQDevice: 0x2353ea0, uuid = 25919FAB-4F89-F299-47A7-F419605354AF, modelName = "Forerunner 965", friendlyName = "Forerunner 965">
Type: Info | Timestamp: 2024-10-08 08:36:12.561448+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c8ac8
Selected device: <IQDevice: 0x2353ea0, uuid = 25919FAB-4F89-F299-47A7-F419605354AF, modelName = "Forerunner 965", friendlyName = "Forerunner 965">
Type: Debug | Timestamp: 2024-10-08 08:36:12.562459+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
GarminServiceTableViewController: Button added
Type: Debug | Timestamp: 2024-10-08 08:36:12.563304+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
GarminServiceTableViewController: Button added
Type: Debug | Timestamp: 2024-10-08 08:36:12.568119+02:00 | Process: Loop | Library: GarminServiceKitUI | Subsystem: Garmin | Category: GarminServiceTableViewController | TID: 0x1c8ac8
GarminService.completeUpdate
Type: Info | Timestamp: 2024-10-08 08:36:14.322908+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c8ac8
Wrote Services to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/Services.plist
Type: Info | Timestamp: 2024-10-08 08:36:14.339133+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c8ac8
Sending most recent glucose: 109.000000 Trend: 4 Date: 2024-10-08 06:34:40 +0000
Type: Debug | Timestamp: 2024-10-08 08:36:17.811059+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c92e6
Sending message: ["glucose": 109.0, "trend": 4, "timestamp": 1728369280]
Type: Info | Timestamp: 2024-10-08 08:36:17.811183+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c92e6
Sent message result: Success
Type: Info | Timestamp: 2024-10-08 08:36:18.769878+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c8ac8
centralManager(_:didDisconnectPeripheral:error:): error=The specified device has disconnected from us. <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = disconnected>
Type: Debug | Timestamp: 2024-10-08 08:37:56.157853+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: BluetoothManager | TID: 0x1c9494
omnipodPeripheralDidDisconnect... will auto-reconnect
Type: Debug | Timestamp: 2024-10-08 08:37:56.160653+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c9494
Reconnecting disconnected autoconnect peripheral
Type: Debug | Timestamp: 2024-10-08 08:37:56.160703+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: BluetoothManager | TID: 0x1c9494
connection (174B525B) Pod disconnected E7E3987E-AD13-3B59-0A71-23D6D549C2FF Optional(Error Domain=CBErrorDomain Code=7 "The specified device has disconnected from us." UserInfo={NSLocalizedDescription=The specified device has disconnected from us.})
Type: Notice | Timestamp: 2024-10-08 08:37:56.164802+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9554
centralManager(_:didConnect:): <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = connected>
Type: Debug | Timestamp: 2024-10-08 08:37:57.333498+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: BluetoothManager | TID: 0x1c9554
PeripheralManager - didConnect: <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = connected>
Type: Debug | Timestamp: 2024-10-08 08:37:57.333617+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9554
PeripheralManager - didConnect - running assertConfiguration
Type: Debug | Timestamp: 2024-10-08 08:37:57.333644+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9554
Configured peripheral has no services. Reconfiguring <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = connected>
Type: Error | Timestamp: 2024-10-08 08:37:57.333789+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
Applying configuration
Type: Notice | Timestamp: 2024-10-08 08:37:57.335486+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
connection (174B525B) Pod connected E7E3987E-AD13-3B59-0A71-23D6D549C2FF
Type: Notice | Timestamp: 2024-10-08 08:37:57.336806+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c956d
didReadRSSI: -63
Type: Notice | Timestamp: 2024-10-08 08:37:57.337961+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9554
didDiscoverServices
Type: Notice | Timestamp: 2024-10-08 08:37:57.541903+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9554
Discovered service: <CBService: 0x3036cb300, isPrimary = YES, UUID = 1A7E4024-E3ED-4464-8B7E-751E03D0DC5F>
Type: Notice | Timestamp: 2024-10-08 08:37:57.542077+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
PodComms completeConfiguration: isPaired=true needsSessionEstablishment=true
Type: Notice | Timestamp: 2024-10-08 08:37:57.572315+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c9494
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:57.737018+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:57.752642+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
Session Established
Type: Debug | Timestamp: 2024-10-08 08:37:58.186493+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c9494
msgSequenceNumber: 3
Type: Info | Timestamp: 2024-10-08 08:37:58.186616+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c9494
Setting podState transport state to ## MessageTransportState
eapSeq: 929
msgSeq: 3
nonceSeq: 0
messageNumber: 2
Type: Debug | Timestamp: 2024-10-08 08:37:58.193029+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c9494
Scheduling session Post-connect status fetch
Type: Notice | Timestamp: 2024-10-08 08:37:58.193645+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.193677+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Delegate configuration notified
Type: Notice | Timestamp: 2024-10-08 08:37:58.193730+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
Peripheral configuration completed
Type: Notice | Timestamp: 2024-10-08 08:37:58.193751+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
======================== Post-connect status fetch ===========================
Type: Notice | Timestamp: 2024-10-08 08:37:58.193810+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
Send(Hex): 174b525b08030e010083aa
Type: Notice | Timestamp: 2024-10-08 08:37:58.195976+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodMessageTransport | TID: 0x1c9494
didSend: 174b525b08030e010083aa
Type: Notice | Timestamp: 2024-10-08 08:37:58.196198+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c9494
send (174B525B) 174b525b08030e010083aa
Type: Notice | Timestamp: 2024-10-08 08:37:58.200256+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c956d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.209982+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.210083+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.214604+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.214689+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.218962+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.219055+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.222817+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.222887+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.226280+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.518066+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Received decrypted response: 174b525b0c0a1d180399000000341fff828e in packet: 302e303d0012174b525b0c0a1d180399000000341fff828e
Type: Debug | Timestamp: 2024-10-08 08:37:58.522642+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodMessageTransport | TID: 0x1c9494
Recv(Hex): 174b525b0c0a1d180399000000341fff828e
Type: Notice | Timestamp: 2024-10-08 08:37:58.522851+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodMessageTransport | TID: 0x1c9494
didReceive: 174b525b0c0a1d180399000000341fff828e
Type: Notice | Timestamp: 2024-10-08 08:37:58.522982+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c9494
receive (174B525B) 174b525b0c0a1d180399000000341fff828e
Type: Notice | Timestamp: 2024-10-08 08:37:58.527013+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9570
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.535883+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.536032+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.541279+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.541372+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9554
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.545456+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9554
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.622151+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c956d
POD Response: StatusResponse(deliveryStatus:Basale Programmata, progressStatus:Normale, timeActive:55h35m, reservoirLevel:50+, insulinDelivered:92.10, bolusNotDelivered:0.00, lastProgrammingMessageSeqNum:0, alerts:Nessun avviso)
Type: Info | Timestamp: 2024-10-08 08:37:58.622515+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodCommsSession | TID: 0x1c9494
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.630750+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c956d
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.630870+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c956d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.637576+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c956d
PumpManager:OmniBLEPumpManager hasNewPumpEvents (lastReconciliation = Optional(2024-10-08 06:37:58 +0000))
Type: Notice | Timestamp: 2024-10-08 08:37:58.637740+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c956d
DU: Stored pod events: []
Type: Info | Timestamp: 2024-10-08 08:37:58.637830+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c956d
Stored doses: []
Type: Info | Timestamp: 2024-10-08 08:37:58.638184+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodCommsSession | TID: 0x1c9494
------------------------ Post-connect status fetch ---------------------------
Type: Notice | Timestamp: 2024-10-08 08:37:58.638287+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
Start of idle at Optional(2024-10-08 06:37:58 +0000)
Type: Notice | Timestamp: 2024-10-08 08:37:58.638360+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9494
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:37:58.638189+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c956d
Thread Performance Checker: Thread running at User-initiated quality-of-service class waiting on a lower QoS thread running at Utility quality-of-service class. Investigate ways to avoid priority inversions
PID: 22517, TID: 1873044
Backtrace
=================================================================
0   OmniBLE                             0x000000010b66ca58 $s7OmniBLE0A14BLEPumpManagerC5store5doses2inSbSayAA15UnfinalizedDoseVG_AA15PodCommsSessionCtF + 260
1   OmniBLE                             0x000000010b68c400 $s7OmniBLE0A14BLEPumpManagerC27podCommsDidEstablishSessionyyAA03PodF0CFyAF0I9RunResultOcfU_SbSayAA15UnfinalizedDoseVGXEfU_ + 64
2   OmniBLE                             0x000000010b55afd4 $s7OmniBLE15PodCommsSessionC15dosesForStorageyySbSayAA15UnfinalizedDoseVGXEF + 288
3   OmniBLE                             0x000000010b68c258 $s7OmniBLE0A14BLEPumpManagerC27podCommsDidEstablishSessionyyAA03PodF0CFyAF0I9RunResultOcfU_ + 752
4   OmniBLE                             0x000000010b58c6ec $s7OmniBLE8PodCommsC10runSession8withName_ySS_yAC0F9RunResultOctFyycfU_ + 1488
5   OmniBLE                             0x000000010b5b3410 $s7OmniBLE17PeripheralManagerC10runSession8withName_ySS_yyctFyyYbcfU_yACcfU_ + 640
6   OmniBLE                             0x000000010b5ace2c $s7OmniBLE17PeripheralManagerC15configureAndRunyyycyACcFyycfU_ + 1820
7   OmniBLE                             0x000000010b4468dc $sIegh_IeyBh_TR + 48
8   libdispatch.dylib                   0x0000000106790a30 _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010679271c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010679a5e8 _dispatch_lane_serial_drain + 828
11  libdispatch.dylib                   0x000000010679b360 _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x00000001067a85f0 _dispatch_root_queue_drain_deferred_wlh + 328
13  libdispatch.dylib                   0x00000001067a7c00 _dispatch_workloop_worker_thread + 580
14  libsystem_pthread.dylib             0x0000000216877c7c _pthread_wqthread + 288
15  libsystem_pthread.dylib             0x0000000216874488 start_wqthread + 8
Type: stdio
Reloading data with context: [Loop.RefreshContext.insulin]
Type: Debug | Timestamp: 2024-10-08 08:37:58.642357+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:37:58.643278+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c956d
Recomputing insulin effects
Type: Debug | Timestamp: 2024-10-08 08:37:58.646691+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9570
Update net basal to Optional(Loop.NetBasal(rate: 0.0, percent: 0.0, start: 2024-10-08 06:34:53 +0000, end: nil))
Type: Debug | Timestamp: 2024-10-08 08:37:58.730387+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c9570
centralManager(_:didConnect:): <CBPeripheral: 0x30189d380, identifier = D90C8504-02F5-B52C-D8EA-CF02CEB1390F, name = Dexcom5L, mtu = 23, state = connected>
Type: Notice | Timestamp: 2024-10-08 08:39:51.855156+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: BluetoothManager | TID: 0x1c96cf
Configured peripheral has no services. Reconfiguring…
Type: Error | Timestamp: 2024-10-08 08:39:51.855311+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: PeripheralManager | TID: 0x1c9861
transmitterDidConnect(_:)
Type: Notice | Timestamp: 2024-10-08 08:39:51.855350+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: TransmitterManager | TID: 0x1c9862
connection (88535L) Connected
Type: Notice | Timestamp: 2024-10-08 08:39:51.858540+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9862
Peripheral configuration completed
Type: Notice | Timestamp: 2024-10-08 08:39:52.061648+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: PeripheralManager | TID: 0x1c9861
Delegate configuration completed
Type: Notice | Timestamp: 2024-10-08 08:39:52.061731+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: PeripheralManager | TID: 0x1c9861
Listening for authentication responses in passive mode
Type: Debug | Timestamp: 2024-10-08 08:39:52.061784+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: Transmitter | TID: 0x1c9861
Ignoring authentication response: 03aebc383aada7311a4b74f617975dfd97
Type: Debug | Timestamp: 2024-10-08 08:39:52.330759+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: Transmitter | TID: 0x1c9861
Observed authenticated session. enabling notifications for control characteristic.
Type: Debug | Timestamp: 2024-10-08 08:39:52.421307+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: Transmitter | TID: 0x1c9861
transmitter(_:didRead:): New glucose
Type: Notice | Timestamp: 2024-10-08 08:39:52.752656+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: TransmitterManager | TID: 0x1c96cf
CGMManager:G6CGMManager did update with newData([LoopKit.NewGlucoseSample(date: 2024-10-08 06:39:40 +0000, quantity: 107 mg/dL, condition: nil, trend: Optional(LoopKit.GlucoseTrend.flat), trendRate: Optional(-0.5 mg/min·dL), isDisplayOnly: false, wasUserEntered: false, syncIdentifier: "88535L 5565374", syncVersion: 1, device: Optional(<<HKDevice: 0x3003c8eb0>, name:CGMBLEKit, manufacturer:Dexcom, model:G6, software:21.0, UDIDeviceIdentifier:00386270000385>))])
Type: Notice | Timestamp: 2024-10-08 08:39:52.753734+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c987c
receive (88535L) New reading: 2024-10-08 06:39:40 +0000
Type: Notice | Timestamp: 2024-10-08 08:39:52.757191+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9862
Purged 1 CachedGlucoseObjects
Type: Notice | Timestamp: 2024-10-08 08:39:52.765512+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: GlucoseStore | TID: 0x1c96cf
<HKObserverQuery E57F4B QoS=Default state=active> notified with changes. Fetching from: <HKQueryAnchor: 1194170 62d7712e5711af8ce70c553edbda7552b90a4238af9b809f878a03cd674185edb581b9714886163b73c7ca9a224fe985ef8f1487b43841c6e8432728da5dabc6>
Type: Notice | Timestamp: 2024-10-08 08:39:52.777103+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: HealthKitSampleStore | TID: 0x1c987c
HKQuery: Executing anchored object query
Type: Notice | Timestamp: 2024-10-08 08:39:52.777554+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: HealthKitSampleStore | TID: 0x1c987c
Stored 1 eligible glucose samples to HealthKit
Type: Notice | Timestamp: 2024-10-08 08:39:52.778641+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: GlucoseStore | TID: 0x1c96cf
Received notification of glucose samples changing
Type: Notice | Timestamp: 2024-10-08 08:39:52.778786+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
RemoteDataType Glucose triggering uploads for: ["Glucose"]
Type: Debug | Timestamp: 2024-10-08 08:39:52.778904+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: RemoteDataServicesManager | TID: 0x1c96cf
[reloadData] from notification with context Optional(Loop.LoopDataManager.LoopUpdateContext.glucose)
Type: Debug | Timestamp: 2024-10-08 08:39:52.780464+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Triggering loop from new CGM data at 2024-10-08 06:39:52 +0000
Type: Notice | Timestamp: 2024-10-08 08:39:52.781003+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9862
Asserting current pump data
Type: Notice | Timestamp: 2024-10-08 08:39:52.781075+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9862
Skipping status update because pumpData is fresh
Type: Notice | Timestamp: 2024-10-08 08:39:52.781355+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c9862
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:39:52.782394+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9882
Reloading data with context: [Loop.RefreshContext.carbs, Loop.RefreshContext.glucose]
Type: Debug | Timestamp: 2024-10-08 08:39:52.783210+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Sending message: ["timestamp": 1728369580, "glucose": 107.0, "trend": 4]
Type: Info | Timestamp: 2024-10-08 08:39:52.783757+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c9881
AnchorQuery results new: 1 deleted: 0 anchor: <HKQueryAnchor: 1194172 62d7712e5711af8ce70c553edbda7552b90a4238af9b809f878a03cd674185edb581b9714886163b73c7ca9a224fe985ef8f1487b43841c6e8432728da5dabc6>
Type: Notice | Timestamp: 2024-10-08 08:39:52.786800+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: HealthKitSampleStore | TID: 0x1c9886
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:39:52.787310+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9882
cgmDataIsStale: false
Type: Debug | Timestamp: 2024-10-08 08:39:52.788855+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: CGMStalenessMonitor | TID: 0x1c8ac8
Updating CGM Staleness timer to fire at 2024-10-08 06:54:40 +0000
Type: Debug | Timestamp: 2024-10-08 08:39:52.788896+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: CGMStalenessMonitor | TID: 0x1c8ac8
Recomputing insulin effects
Type: Debug | Timestamp: 2024-10-08 08:39:52.791342+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Sample 6D0FA59A-076B-43D8-AB5E-8C64BFA74C4F from HKAnchoredObjectQuery already present in cache
Type: Notice | Timestamp: 2024-10-08 08:39:52.825828+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: GlucoseStore | TID: 0x1c9881
Fetching counteraction effects after 2024-10-08 06:34:40 +0000
Type: Debug | Timestamp: 2024-10-08 08:39:52.828901+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Recomputing prediction and recommendations.
Type: Debug | Timestamp: 2024-10-08 08:39:52.849542+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
 *** Glucose: 107 mg/dL, effectiveBolusApplicationFactor: 0.40
Type: Debug | Timestamp: 2024-10-08 08:39:52.850819+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Rounded 0.0 to 0.0
Type: Notice | Timestamp: 2024-10-08 08:39:52.852689+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9862
Rounded 10.0 to 10.0
Type: Notice | Timestamp: 2024-10-08 08:39:52.852948+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9862
No dose recommended.
Type: Notice | Timestamp: 2024-10-08 08:39:52.852967+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Looping with timeBasedDoseApplicationFactor = 0.9990267964204153
Type: Notice | Timestamp: 2024-10-08 08:39:52.854624+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Loop running
Type: Notice | Timestamp: 2024-10-08 08:39:52.854645+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Alert not found for update: Loop.significantTimeChange
Type: Error | Timestamp: 2024-10-08 08:39:52.855176+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: AlertStore | TID: 0x1c96cf
Loop completed successfully.
Type: Notice | Timestamp: 2024-10-08 08:39:52.879140+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Loop success Optional([AnyHashable("duration"): 0.02444899082183838])
Type: Debug | Timestamp: 2024-10-08 08:39:52.879264+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: AnalyticsServicesManager | TID: 0x1c9862
Loop ended
Type: Notice | Timestamp: 2024-10-08 08:39:52.883356+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Version check failed for LoopSupportUI
Type: Error | Timestamp: 2024-10-08 08:39:52.883624+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: SupportManager | TID: 0x1c9886
[reloadData] from notification with context Optional(Loop.LoopDataManager.LoopUpdateContext.loopFinished)
Type: Debug | Timestamp: 2024-10-08 08:39:52.884021+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Purged 2 DosingDecisionObjects
Type: Info | Timestamp: 2024-10-08 08:39:52.888548+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9861
RemoteDataType DosingDecision triggering uploads for: ["DosingDecision"]
Type: Debug | Timestamp: 2024-10-08 08:39:52.888634+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: RemoteDataServicesManager | TID: 0x1c9861
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438904): queueWait(0.002), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:52.891365+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9881
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438904): queueWait(0.001), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:52.893187+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9881
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438904): queueWait(0.013), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:52.910840+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9881
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438904): queueWait(0.000), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:52.911135+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9881
Update net basal to Optional(Loop.NetBasal(rate: 0.0, percent: 0.0, start: 2024-10-08 06:34:53 +0000, end: nil))
Type: Debug | Timestamp: 2024-10-08 08:39:52.915726+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c9862
[reloadData] due to context change during previous reload
Type: Debug | Timestamp: 2024-10-08 08:39:52.920705+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Reloading data with context: [Loop.RefreshContext.insulin]
Type: Debug | Timestamp: 2024-10-08 08:39:52.921600+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Rounded 0.0 to 0.0
Type: Notice | Timestamp: 2024-10-08 08:39:52.986813+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9862
Manual bolus rec = Optional(LoopKit.ManualBolusRecommendationWithDate(recommendation: LoopKit.ManualBolusRecommendation(amount: 0.0, pendingInsulin: 0.0, notice: Optional(LoopKit.BolusRecommendationNotice.predictedGlucoseInRange)), date: 2024-10-08 06:39:52 +0000))
Type: Debug | Timestamp: 2024-10-08 08:39:52.987034+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopDataManager | TID: 0x1c9862
Purged 0 DosingDecisionObjects
Type: Info | Timestamp: 2024-10-08 08:39:52.992502+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9884
RemoteDataType DosingDecision triggering uploads for: ["DosingDecision"]
Type: Debug | Timestamp: 2024-10-08 08:39:52.992565+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: RemoteDataServicesManager | TID: 0x1c9884
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438905): queueWait(0.014), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:53.006743+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c96cf
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438905): queueWait(0.000), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:53.007197+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c96cf
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438905): queueWait(0.000), fetch(0.000)
Type: Debug | Timestamp: 2024-10-08 08:39:53.009964+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9883
Update net basal to Optional(Loop.NetBasal(rate: 0.0, percent: 0.0, start: 2024-10-08 06:34:53 +0000, end: nil))
Type: Debug | Timestamp: 2024-10-08 08:39:53.011925+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c9862
centralManager(_:didDisconnectPeripheral:error:): <CBPeripheral: 0x30189d380, identifier = D90C8504-02F5-B52C-D8EA-CF02CEB1390F, name = Dexcom5L, mtu = 23, state = disconnected>
Type: Notice | Timestamp: 2024-10-08 08:39:53.057023+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: BluetoothManager | TID: 0x1c9862
executeDosingDecisionQuery (anchor = QueryAnchor(modificationCounter: 438905): queueWait(0.000), fetch(0.001)
Type: Debug | Timestamp: 2024-10-08 08:39:53.498367+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: DosingDecisionStore | TID: 0x1c9883
Sent message result: Success
Type: Info | Timestamp: 2024-10-08 08:39:53.815146+02:00 | Process: Loop | Library: GarminServiceKit | Subsystem: Garmin | Category: GarminService | TID: 0x1c8ac8
Re-connecting to known peripheral D90C8504-02F5-B52C-D8EA-CF02CEB1390F
Type: Debug | Timestamp: 2024-10-08 08:39:55.120273+02:00 | Process: Loop | Library: CGMBLEKit | Subsystem: com.loopkit.CGMBLEKit | Category: BluetoothManager | TID: 0x1c9881
Refreshing widget. Reason: Loop completed
Type: Notice | Timestamp: 2024-10-08 08:39:58.124747+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: LoopWidgets | TID: 0x1c8ac8
centralManager(_:didDisconnectPeripheral:error:): error=The specified device has disconnected from us. <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = disconnected>
Type: Debug | Timestamp: 2024-10-08 08:41:01.810332+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: BluetoothManager | TID: 0x1c99a3
omnipodPeripheralDidDisconnect... will auto-reconnect
Type: Debug | Timestamp: 2024-10-08 08:41:01.810806+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c99a3
Reconnecting disconnected autoconnect peripheral
Type: Debug | Timestamp: 2024-10-08 08:41:01.810861+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: BluetoothManager | TID: 0x1c99a3
connection (174B525B) Pod disconnected E7E3987E-AD13-3B59-0A71-23D6D549C2FF Optional(Error Domain=CBErrorDomain Code=7 "The specified device has disconnected from us." UserInfo={NSLocalizedDescription=The specified device has disconnected from us.})
Type: Notice | Timestamp: 2024-10-08 08:41:01.814156+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9a2d
centralManager(_:didConnect:): <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = connected>
Type: Debug | Timestamp: 2024-10-08 08:41:02.976685+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: BluetoothManager | TID: 0x1c9a2d
PeripheralManager - didConnect: <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = connected>
Type: Debug | Timestamp: 2024-10-08 08:41:02.976857+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9a2d
PeripheralManager - didConnect - running assertConfiguration
Type: Debug | Timestamp: 2024-10-08 08:41:02.976902+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9a2d
Configured peripheral has no services. Reconfiguring <CBPeripheral: 0x3018989c0, identifier = E7E3987E-AD13-3B59-0A71-23D6D549C2FF, name = TWI BOARD, mtu = 23, state = connected>
Type: Error | Timestamp: 2024-10-08 08:41:02.977435+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
connection (174B525B) Pod connected E7E3987E-AD13-3B59-0A71-23D6D549C2FF
Type: Notice | Timestamp: 2024-10-08 08:41:02.981129+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9a2d
Applying configuration
Type: Notice | Timestamp: 2024-10-08 08:41:02.981262+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
didReadRSSI: -62
Type: Notice | Timestamp: 2024-10-08 08:41:02.981415+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9a2d
didDiscoverServices
Type: Notice | Timestamp: 2024-10-08 08:41:03.185128+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c9a2d
Discovered service: <CBService: 0x3039a4140, isPrimary = YES, UUID = 1A7E4024-E3ED-4464-8B7E-751E03D0DC5F>
Type: Notice | Timestamp: 2024-10-08 08:41:03.185339+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
PodComms completeConfiguration: isPaired=true needsSessionEstablishment=true
Type: Notice | Timestamp: 2024-10-08 08:41:03.217500+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c99a3
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:03.367669+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:03.384126+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
Session Established
Type: Debug | Timestamp: 2024-10-08 08:41:03.785928+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c99a3
msgSequenceNumber: 3
Type: Info | Timestamp: 2024-10-08 08:41:03.786043+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c99a3
Setting podState transport state to ## MessageTransportState
eapSeq: 930
msgSeq: 3
nonceSeq: 0
messageNumber: 4
Type: Debug | Timestamp: 2024-10-08 08:41:03.786135+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodComms | TID: 0x1c99a3
Scheduling session Post-connect status fetch
Type: Notice | Timestamp: 2024-10-08 08:41:03.786834+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:03.786861+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a82
Delegate configuration notified
Type: Notice | Timestamp: 2024-10-08 08:41:03.786965+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
Peripheral configuration completed
Type: Notice | Timestamp: 2024-10-08 08:41:03.787070+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
======================== Post-connect status fetch ===========================
Type: Notice | Timestamp: 2024-10-08 08:41:03.787110+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
Send(Hex): 174b525b10030e0100038c
Type: Notice | Timestamp: 2024-10-08 08:41:03.788398+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodMessageTransport | TID: 0x1c99a3
didSend: 174b525b10030e0100038c
Type: Notice | Timestamp: 2024-10-08 08:41:03.788535+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c99a3
send (174B525B) 174b525b10030e0100038c
Type: Notice | Timestamp: 2024-10-08 08:41:03.796508+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9a2d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:03.808621+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a82
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:03.809183+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a82
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:03.815797+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a82
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:03.815895+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a82
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:03.820539+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a82
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:03.820622+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a82
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:03.824790+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a82
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:03.824873+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a82
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:03.827682+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a82
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:04.132228+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
Received decrypted response: 174b525b140a1d180399800000342bff80f4 in packet: 302e303d0012174b525b140a1d180399800000342bff80f4
Type: Debug | Timestamp: 2024-10-08 08:41:04.138148+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodMessageTransport | TID: 0x1c99a3
Recv(Hex): 174b525b140a1d180399800000342bff80f4
Type: Notice | Timestamp: 2024-10-08 08:41:04.138364+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodMessageTransport | TID: 0x1c99a3
didReceive: 174b525b140a1d180399800000342bff80f4
Type: Notice | Timestamp: 2024-10-08 08:41:04.138477+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c99a3
receive (174B525B) 174b525b140a1d180399800000342bff80f4
Type: Notice | Timestamp: 2024-10-08 08:41:04.142499+02:00 | Process: Loop | Library: LoopKit | Subsystem: com.loopkit.LoopKit | Category: PersistentDeviceLog | TID: 0x1c9a82
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:04.148910+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:04.149052+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:04.155000+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:04.155086+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:04.159097+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:04.236501+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
POD Response: StatusResponse(deliveryStatus:Basale Programmata, progressStatus:Normale, timeActive:55h38m, reservoirLevel:50+, insulinDelivered:92.15, bolusNotDelivered:0.00, lastProgrammingMessageSeqNum:0, alerts:Nessun avviso)
Type: Info | Timestamp: 2024-10-08 08:41:04.236787+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodCommsSession | TID: 0x1c99a3
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:04.244778+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:04.244889+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:04.251354+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
PumpManager:OmniBLEPumpManager hasNewPumpEvents (lastReconciliation = Optional(2024-10-08 06:41:04 +0000))
Type: Notice | Timestamp: 2024-10-08 08:41:04.251671+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
DU: Stored pod events: []
Type: Info | Timestamp: 2024-10-08 08:41:04.251747+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: OmniBLEPumpManager | TID: 0x1c9a2d
PumpManager:OmniBLEPumpManager did update state
Type: Notice | Timestamp: 2024-10-08 08:41:04.252062+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: DeviceDataManager | TID: 0x1c9a2d
Stored doses: []
Type: Info | Timestamp: 2024-10-08 08:41:04.252080+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: PodCommsSession | TID: 0x1c99a3
------------------------ Post-connect status fetch ---------------------------
Type: Notice | Timestamp: 2024-10-08 08:41:04.252194+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
Start of idle at Optional(2024-10-08 06:41:04 +0000)
Type: Notice | Timestamp: 2024-10-08 08:41:04.252258+02:00 | Process: Loop | Library: OmniBLE | Subsystem: com.randallknutson.OmniBLE | Category: DashPeripheralManager | TID: 0x1c99a3
Thread Performance Checker: Thread running at User-initiated quality-of-service class waiting on a lower QoS thread running at Utility quality-of-service class. Investigate ways to avoid priority inversions
PID: 22517, TID: 1874339
Backtrace
=================================================================
0   OmniBLE                             0x000000010b66ca58 $s7OmniBLE0A14BLEPumpManagerC5store5doses2inSbSayAA15UnfinalizedDoseVG_AA15PodCommsSessionCtF + 260
1   OmniBLE                             0x000000010b68c400 $s7OmniBLE0A14BLEPumpManagerC27podCommsDidEstablishSessionyyAA03PodF0CFyAF0I9RunResultOcfU_SbSayAA15UnfinalizedDoseVGXEfU_ + 64
2   OmniBLE                             0x000000010b55afd4 $s7OmniBLE15PodCommsSessionC15dosesForStorageyySbSayAA15UnfinalizedDoseVGXEF + 288
3   OmniBLE                             0x000000010b68c258 $s7OmniBLE0A14BLEPumpManagerC27podCommsDidEstablishSessionyyAA03PodF0CFyAF0I9RunResultOcfU_ + 752
4   OmniBLE                             0x000000010b58c6ec $s7OmniBLE8PodCommsC10runSession8withName_ySS_yAC0F9RunResultOctFyycfU_ + 1488
5   OmniBLE                             0x000000010b5b3410 $s7OmniBLE17PeripheralManagerC10runSession8withName_ySS_yyctFyyYbcfU_yACcfU_ + 640
6   OmniBLE                             0x000000010b5ace2c $s7OmniBLE17PeripheralManagerC15configureAndRunyyycyACcFyycfU_ + 1820
7   OmniBLE                             0x000000010b4468dc $sIegh_IeyBh_TR + 48
8   libdispatch.dylib                   0x0000000106790a30 _dispatch_call_block_and_release + 32
9   libdispatch.dylib                   0x000000010679271c _dispatch_client_callout + 20
10  libdispatch.dylib                   0x000000010679a5e8 _dispatch_lane_serial_drain + 828
11  libdispatch.dylib                   0x000000010679b360 _dispatch_lane_invoke + 408
12  libdispatch.dylib                   0x00000001067a85f0 _dispatch_root_queue_drain_deferred_wlh + 328
13  libdispatch.dylib                   0x00000001067a7c00 _dispatch_workloop_worker_thread + 580
14  libsystem_pthread.dylib             0x0000000216877c7c _pthread_wqthread + 288
15  libsystem_pthread.dylib             0x0000000216874488 start_wqthread + 8
Type: stdio
Reloading data with context: [Loop.RefreshContext.insulin]
Type: Debug | Timestamp: 2024-10-08 08:41:04.257255+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c8ac8
Wrote PumpManagerState to file:///var/mobile/Containers/Data/Application/62308BA8-18A0-40B6-ACDF-1F2117939158/Documents/PumpManagerState.plist
Type: Info | Timestamp: 2024-10-08 08:41:04.258544+02:00 | Process: Loop | Library: LoopCore | TID: 0x1c9a2d
Update net basal to Optional(Loop.NetBasal(rate: 0.0, percent: 0.0, start: 2024-10-08 06:34:53 +0000, end: nil))
Type: Debug | Timestamp: 2024-10-08 08:41:04.320183+02:00 | Process: Loop | Library: Loop | Subsystem: com.loopkit.Loop | Category: StatusTableViewController | TID: 0x1c9a82
```

**Comment by flavi0gritti** [comment]
Here's a txt file I created with the log content
[xcodelog.txt](https://github.com/user-attachments/files/17289059/xcodelog.txt)

**Comment by flavi0gritti** [comment]
I've done some testing and I think I found the culprit, within the CGM menu in the loop app there is an "upload toggle" that I wasn't aware off before setting up nightscout and I am now noticing the with the toggle enabled the phone and watch are perfectly in sync!

**Comment by flavi0gritti** [comment]
With it off, the devices lost connection as soon as the loop app glucose value updates and here's the log of it
[xcodelog_error.txt](https://github.com/user-attachments/files/17289155/xcodelog_error.txt)

Will test with the edge 1040 as soon as I have some time during work hours

**Comment by flavi0gritti** [comment]
Can confirm that with "upload reading" toggled on in the cgm menu within the loop app the connection works perfectly fine while it doesn't update at all with it disabled.

Should I close this issue now? I don't think it's an issue with the Garmin service you developed, maybe it could be added as a must do within the repo readme.

**Comment by janvv** [comment]
Copy from Zulip here:

@flavi0gritti You are absolutely right. I forgot to add this to the setup instructions and didn't think about it anymore. You're a testing genius :muscle:

Let me explain: The Service derives from a remote data service class which is designed to upload data to a remote locations. Such as the nightscout service, which also derives from that class. However, services of that class won't receive updates from CGMs if the upload toggle is off.

However, upon selecting the garmin device, i fetch the local apple health database to transmit the latest glucose entry. This is why the first value appears but then receives no further updates as long as the toggle is "off".

I will update the documentation. Then, close the issue. Also, very happy to see that it is now working for you.

I can already see that this will become a bottle neck for those users that don't want to upload their CGM data to nightscout but still use the Garmin service. I created a new issue for it: #5

**Comment by janvv** [comment]
I [Updated](https://github.com/janvv/GarminService/commit/8918e2a7e00480b7e0a049b914be335575a7b9df) the README. Closing the issue now.
